amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
This seems fine, but is there a case where JUJU_TEST_CONSTRAINTS should be merged with the constraints provided by the author? Either to override or append?